crypto/decrepit/cfb/cfb.c

```diff
@@ -36,6 +40,55 @@ static int aes_cfb_init_key(EVP_CIPHER_CTX *ctx, const uint8_t *key,
   return 1;
 }
 
+static int aes_cfb1_cipher(EVP_CIPHER_CTX *ctx, uint8_t *out,
+                            const uint8_t *in, size_t len) {
+  if (!out || !in) {
+    return 0;
+  }
+
+  EVP_CFB_CTX *cfb_ctx = ctx->cipher_data;
+  if (ctx->flags & EVP_CIPH_FLAG_LENGTH_BITS) {
+    int num = ctx->num;
+    AES_cfb1_encrypt(in, out, len, &cfb_ctx->ks, ctx->iv, &num,
+                      ctx->encrypt ? AES_ENCRYPT : AES_DECRYPT);
+    ctx->num = num;
```

Comment:
Wondering why this assigned is made? `AES_cfb{1,8}_{encrypt, decrypt}()` doesn't use this parameter.

Reply:
It's used and passed to its downstream. Eventually, used in `|cfbr_encrypt_block|`.

Reply:
`ctx->encrypt ? AES_ENCRYPT : AES_DECRYPT` is passed as `enc` to `|CRYPTO_cfb128_1_encrypt|`, and then `|cfbr_encrypt_block|`..

```
if (enc) {
    // encrypt the input
    for (n = 0; n < num; ++n) {
      out[n] = (ovec[16 + n] = in[n] ^ ivec[n]);
    }
  } else {
    // decrypt the input
    for (n = 0; n < num; ++n) {
      out[n] = (ovec[16 + n] = in[n]) ^ ivec[n];
    }
  }
```

If this parameter is not passed, how does the cipher know the direction(encryption/decryption)?

Reply:
I'm talking about the `num` parameter, not the `enc` parameter :)

Reply:
I thought we talked num in https://github.com/awslabs/aws-lc/pull/595#discussion_r955606572. I noticed this is not used but not sure why the comment was written in that way. I considered removing this but decided to keep it in case I missed some nuance(e.g. commit 2, while unlikely).